javaslang/generator/Generator.scala

```diff
@@ -51,6 +51,302 @@ def generateMainClasses(): Unit = {
+                  return Function$i.of(methodReference);
+              }
+            """
+        })("\n\n")}
+
+          ${(0 to N).gen(i => {
+          val generics = (1 to i).gen(j => s"T$j")(", ")
+          val fullGenerics = s"<${(i > 0).gen(s"$generics, ")}R>"
+          xs"""
+              /$javadoc
+               * Alias for {@code CheckedFunction$i.of()}
+               *
+               * @see CheckedFunction$i#of(CheckedFunction$i)
+               */
+              public static $fullGenerics CheckedFunction$i$fullGenerics cf$i(CheckedFunction$i$fullGenerics methodReference) {
+                  return CheckedFunction$i.of(methodReference);
+              }
+            """
+        })("\n\n")}
+
+          /$javadoc
+           * Alias for {@code Tuple.empty()}
+           *
+           * @see Tuple#empty()
+           */
+          public static Tuple0 tuple() {
+            return Tuple.empty();
+          }
+
+          ${(1 to N).gen(i => {
+          val generics = (1 to i).gen(j => s"T$j")(", ")
+          val params = (1 to i).gen(j => s"T$j t$j")(", ")
+          val args = (1 to i).gen(j => s"t$j")(", ")
+          xs"""
+              /$javadoc
+               * Alias for {@code Tuple.of($generics)}
+               *
+               * @see Tuple#of($generics)
+               */
+              public static <$generics> Tuple$i<$generics> tuple($params) {
+                  return Tuple.of($args);
+              }
+            """
+        })("\n\n")}
+
+          /$javadoc
+           * Alias for {@code $EitherType.right()}
+           *
+           * @see $EitherType#right(R)
+           */
+          public static <L, R> $EitherType<L, R> right(R right) {
+            return $EitherType.right(right);
+          }
+
+          /$javadoc
+           * Alias for {@code $EitherType.left()}
+           *
+           * @see $EitherType#left(R)
+           */
+          public static <L, R> $EitherType<L, R> left(L left) {
+            return $EitherType.left(left);
+          }
+
+          /$javadoc
+           * Alias for {@code $FutureType.of()}
+           *
+           * @see $FutureType#of($CheckedSupplierType)
+           */
+          public static <T> $FutureType<T> future($CheckedSupplierType<? extends T> computation) {
+            return $FutureType.of(computation);
+          }
+
+          /$javadoc
+           * Alias for {@code $FutureType.of()}
+           *
+           * @see $FutureType#of($ExecutorServiceType, $CheckedSupplierType)
+           */
+          public static <T> $FutureType<T> future($ExecutorServiceType executorService, $CheckedSupplierType<? extends T> computation) {
+            return $FutureType.of(executorService, computation);
+          }
+
+          /$javadoc
+           * Alias for {@code $FutureType.successful()}
+           *
+           * @see $FutureType#successful(T)
+           */
+          public static <T> $FutureType<T> future(T result) {
```

Comment:
I think `future(Throwable)` and `future(T)` are ambiguous. Please try if the compiler knows what to do when using `future(new Error("oh"))`